### DISCUSSION TITLE
What is the best way to have a bidirectional/clickable 'reference' on the FROM document?

### DISCUSSION THREAD

**Discussion by gobbletown** [jglev/obsidian-carry-forward#21] [Q&A]
I would like to have a clickable link or bidirectional reference when using Carry-Forward.

That is:

Note A ((SOURCE or FROM Note))
dolorem ipsum, meh ^[[SOME TEXT pointing to Note B]] 

Note B ((REFERRING or TO Note))
dolorem ipsum, meh ^[[SOME TEXT referencing Note A]]

***
Where [[SOME TEXT]] is user defined and there is a clickable reference (or icon or * or some such)  instead of an invisible, unidirectional reference.

Or, desired effects:
A way to see in **Note A** any references to **otherNotes** that allow travel from **SOURCE Note** to **REFERRING Note** with a click n' hover.

I feel like this is likely a feature present already...  Just having trouble connecting dots...

Thansk!

**Comment by jglev** [comment]
@gobbletown, thank you for writing!

Would you be willing to say a bit more about the use-case you have in mind (I.e., with what type of workflow you envision you would use this feature)? I think that that would help me to understand and give a more nuanced response. Thank you!

**Reply by gobbletown** [reply]
Surely!

In short, I'm looking for a way that allows for annotating original works with my own thoughts. I'd like a selection of original text to open in a new note that contains my ideas on that original text. I'd like a clickable link within the original text to take me to my annotated note and a clickable link within the annotated note that takes me back to the specific line of text in the original document.

Let's say I'm reading the text of a long document within my obsidian vault. 

I'd like to be able to select text in the original document to be pulled into a new note that contains my thoughts on that specific line. When I'm reading/re-reading the original text later, I'd like to be able to see which lines have associated annotated content and then be able to click on links to take me to those annotations and from the annotations back again.

Does that help?

Basically an in-line bidirectional link from a block of original content to its annotation and back. 

Is that doable?

**Reply by jglev** [reply]
Thank you for explaining more!

> I feel like this is likely a feature present already... Just having trouble connecting dots...

There are two existing ways that come to mind for displaying (and previewing with `Ctrl + hover`) all locations where a given line has been linked:

1. In the source note, highlighting the line ID at the end of the line (`^12345`) and running `Search: Search in all files` will show all locations across all files that link to that line. `Ctrl + hover` works with these results.  
    ![Screenshot_20221105_084333](https://user-images.githubusercontent.com/3667562/200120419-52ddba39-4f28-418c-bbc1-359adc9c2c4b.png)

1. The "Linked mentions" pane will show all linked mentions back to the current file (albeit not specific to the current line one is on). As above, `Ctrl + hover` works here, as well.  
    ![Screenshot_20221105_084234](https://user-images.githubusercontent.com/3667562/200120379-628cb181-8044-4e43-822c-da0f4590550b.png)

Do either of those match with what you're envisioning?
